velox/functions/prestosql/Map.cpp

```diff
@@ -13,367 +13,15 @@
-      context.moveOrCopyResult(mapVector, *remainingRows, result);
-    }
-  }
-
-  static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
-    // array(K), array(V) -> map(K,V)
-    // () -> map()
-    return {
-        exec::FunctionSignatureBuilder()
-            .typeVariable("K")
-            .typeVariable("V")
-            .returnType("map(K,V)")
-            .argumentType("array(K)")
-            .argumentType("array(V)")
-            .build(),
-        exec::FunctionSignatureBuilder()
-            .returnType("map(unknown,unknown)")
-            .build(),
-    };
-  }
-
- private:
-  // Can only take the fast path if keys and values have an equal
-  // number of arrays and the offsets and sizes of these arrays match
-  // 1:1. The map must be well formed for all elements, also ones not
-  // in 'rows' in apply(). This is because canonicalize() will touch
-  // all elements in any case.
-  bool canTakeFastPath(
-      ArrayVector* keys,
-      ArrayVector* values,
-      const SelectivityVector& rows) const {
-    VELOX_CHECK_GE(keys->size(), rows.end());
-    VELOX_CHECK_GE(values->size(), rows.end());
-    // the fast path takes a reference to the keys and values and the
-    // offsets and sizes from keys. This is valid only if the keys and
-    // values align for all rows for both size and offset. Anything
-    // else will break canonicalize().
-    if (keys->size() != values->size()) {
-      return false;
-    }
-    for (auto row = 0; row < keys->size(); ++row) {
-      if (keys->isNullAt(row)) {
-        continue;
-      }
-
-      if (values->isNullAt(row)) {
-        return false;
-      }
-
-      if (keys->offsetAt(row) != values->offsetAt(row) ||
-          keys->sizeAt(row) != values->sizeAt(row)) {
-        return false;
-      }
-    }
-    return true;
-  }
-
-  std::optional<vector_size_t> findDuplicateKeys(
-      const std::vector<vector_size_t>& sortedIndices,
-      const VectorPtr& keysElements) const {
-    for (auto i = 1; i < sortedIndices.size(); ++i) {
-      if (keysElements->equalValueAt(
-              keysElements.get(), sortedIndices[i], sortedIndices[i - 1])) {
-        return sortedIndices[i];
-      }
-    }
-    return std::nullopt;
-  }
-
-  void checkDuplicateConstantKeys(
-      const std::vector<vector_size_t>& sortedIndices,
-      const VectorPtr& keysElements) const {
-    static const char* kDuplicateKey =
-        "Duplicate map keys ({}) are not allowed";
-
-    if (auto duplicateIndex = findDuplicateKeys(sortedIndices, keysElements)) {
-      auto duplicateKey = keysElements->wrappedVector()->toString(
-          keysElements->wrappedIndex(duplicateIndex.value()));
-      VELOX_USER_FAIL(kDuplicateKey, duplicateKey);
-    }
-  }
-};
-} // namespace
 
 VELOX_DECLARE_VECTOR_FUNCTION(
     udf_map,
-    MapFunction</*AllowDuplicateKeys=*/false>::signatures(),
+    MapFunction<false>::signatures(),
     std::make_unique<MapFunction</*AllowDuplicateKeys=*/false>>());
```

Comment:
The AllowDuplicateKeys value and DeduplicateKeys values are always same, why add two arguments? Can we remove DeduplicateKeys?

Reply:
In Presto, `DeduplicateKeys` should be always false. In Spark, it should be true when `AllowDuplicateKeys` is true